src/components/select/Select.js

```diff
@@ -97,9 +97,12 @@ export class SelectComponent extends BaseComponent {
     if (typeof data === 'string') {
       return this.t(data);
     }
+
     const template = this.component.template ? this.interpolate(this.component.template, {item: data}) : data.label;
-    const label = template.replace(/<\/?[^>]+(>|$)/g, '');
-    return template.replace(label, this.t(label));
+    if (template) {
+      const label = template.replace(/<\/?[^>]+(>|$)/g, '');
+      return template.replace(label, this.t(label));
+    }
```

Comment:
So, does this mean that if there isn't any template it will return ```undefined```. This would probably show ```undefined``` in the datagrid. Is this expected?

Reply:
Datagrid is blank and no value is recorded in the <td></td>
https://www.screencast.com/t/bFgmNdRv

Reply:
Ok, but what about Select components that do not define templates (like manually configured), but also have data as an object. This will return empty which isn't right. At that point, I would prefer to do something like this just so that they get a representation of what data is stored.

```
else {
  return JSON.stringify(data);
}
```

Reply:
Also, I don't see anywhere that is checking to see if the data is ```null``` to return an empty string.  It needs this at the beginning.

```
if (!data) {
  return '';
}
```